### ISSUE TITLE
Could not build module 'ResponseDetective' 

### ISSUE THREAD

**Issue by SandeepAggarwal** [netguru/ResponseDetective#21]
In line
`@import ResponseDetective; ` 
I am getting following compile time error in Obj-C for branch **feature/swift-3-dev**
**Could not build module 'ResponseDetective'**

**Comment by akashivskyy** [comment]
How are you adding ResponseDetective to your project?

**Comment by SandeepAggarwal** [comment]
Using Cocoapods 
`pod 'ResponseDetective', :git => 'https://github.com/netguru/ResponseDetective.git/' , :branch => 'feature/swift-3-dev'`

**Comment by akashivskyy** [comment]
Does `#import <ResponseDetective/ResponseDetective.h>` work?

**Comment by SandeepAggarwal** [comment]
Yes, `#import <ResponseDetective/ResponseDetective.h>` it gets complile but doesn't serve any purpose.
Then I get an error saying 

> Declaration of 'RDTResponseDetective' must be imported from module 'ResponseDetective.Swift' before it is required

Hence I go with `@import ResponseDetective;`

**Comment by akashivskyy** [comment]
Ok, I will take a look at this and try to reproduce.

**Comment by jaymathew** [comment]
Same issue here

**Comment by VladKorzun** [comment]
Same issue when trying to import 0.5 version in Objective-C project

**Comment by akashivskyy** [comment]
@VladKorzun @jaymathew: Can you paste here the exact error you're seeing?

**Comment by VladKorzun** [comment]
I've installed ResponseDetective via cocoapods:

`pod 'ResponseDetective', '~> 0.5'`

Then when I try to import it

`@import ResponseDetective;`

I get "Could not build module 'ResponseDetective'" error. If I try to import it like

`#import <ResponseDetective/ResponseDetective.h>`

and try to enable configuration 

`[RDTResponseDetective enableInConfiguration:configuration];`

I get "Declaration of 'RDTResponseDetective' must be imported from module 'ResponseDetective.Swift' before it is required"  error

**Comment by akashivskyy** [comment]
@VladKorzun: Can you also paste the output of `xcodebuild -version`?

**Comment by VladKorzun** [comment]
Xcode 8.2.1
Build version 8C1002

**Comment by akashivskyy** [comment]
And does your project contain any Swift files or is it Objective-C only? (I'm just trying to reproduce the issue).

**Comment by VladKorzun** [comment]
Objective-C only

**Comment by akashivskyy** [comment]
Are you not seeing "Convert to Current Swift Syntax?" dialog after installing ResponseDetective?

**Comment by VladKorzun** [comment]
Yes! Thank you, I've got it. It was just unclear for me from errors I've got that the issue was caused by unspecified SWIFT_VERSION. 

Thanks!

**Comment by akashivskyy** [comment]
Okay, it seems I identified the issue. **TL;DR: Will be fixed in ResponseDetective 1.0.**

When integrating with Objective-C-only projects, CocoaPods doesn't know which Swift version to use and doesn't set `SWIFT_VERSION` build setting for Swift pods.

[Starting with CocoaPods 1.1.0](http://blog.cocoapods.org/CocoaPods-1.1.0/), there is a [`pushed_with_swift_version`](https://github.com/CocoaPods/Specs/blob/master/Specs/8/0/9/Quick/1.1.0/Quick.podspec.json#L41) podspec JSON key that CocoaPods uses when installing a pod, but it only works for already pushed versions (and ResponseDetective 0.5 was pushed before that).

In addition, CocoaPods generates its own umbrella header file, `ResponseDetective-umbrella.h`, ignoring the already existing `ResponseDetective.h`. This causes the error reported by @SandeepAggarwal, because Swift types are not available after importing a non-umbrella header. To fix that, [I pushed a custom `modulemap`](03a2f079afde27b6c95c144100959a70d77e3aaa) and told CocoaPods to use that.

The two above problems are now fixed and will be included in ResponseDetective 1.0, which will be released today.
